core/emulate.c

```diff
@@ -854,10 +905,49 @@ em_status_t EMCALL em_decode_insn(struct em_context_t *ctxt, const uint8_t *insn
             b = insn_fetch_u8(ctxt);
             *opcode = opcode_table_0F3A[b];
             break;
-
         default:
             *opcode = opcode_table_0F[b];
         }
+        break;
+
+    case 0xC4:
+    case 0xC5:
+        /* Intel SDM Vol. 2A: 2.3.5 The VEX Prefix */
+        ctxt->vex.prefix = b;
+        if (b == 0xC4) {
+            ctxt->vex.value = insn_fetch_u16(ctxt);
+        } else {
+            ctxt->vex.value = insn_fetch_u8(ctxt);
+            ctxt->vex.r = ctxt->vex.w;
+            ctxt->vex.w = 1;
+            ctxt->vex.x = 1;
+            ctxt->vex.b = 1;
+            ctxt->vex.m = 1;
+        }
+        if (ctxt->mode != EM_MODE_PROT64 && (!ctxt->vex.r || !ctxt->vex.x)) {
+            return EM_ERROR;
```

Comment:
nit: According to Intel SDM Vol. 2A 2.3.5.{2, 3, 6}, in this case we should reinterpret `vex.prefix` as either the `LDS` (0xC5) or `LES` (0xC4) instruction. We don't have to support emulating these two uncommon instructions (although, in theory, they may do MMIO), but it would be useful to add a comment to explain the situation or leave it as a TODO.